cpp/velox/compute/WholeStageResultIterator.cc

```diff
@@ -307,6 +308,18 @@ void WholeStageResultIterator::collectMetrics() {
       metrics_->get(Metrics::kIoWaitTime)[metricIndex] = runtimeMetric("sum", second->customStats, kIoWaitTime);
       metrics_->get(Metrics::kPreloadSplits)[metricIndex] =
           runtimeMetric("sum", entry.second->customStats, kPreloadSplits);
+
+      // Update the Write metrics.
+      for (int i = 0; i < task_->taskStats().pipelineStats.size(); ++i) {
```

Comment:
why do we need two extra loop ? can not be `if (entry.first == "TableWrite") {...}` ?

Reply:
@ulysses-you The metrics for `TableWrite `are recorded in the [OperatorStats ](https://github.com/facebookincubator/velox/blob/e907a80013fddca7b38f769b4168a1d7ca1510f6/velox/exec/TableWriter.cpp#L246) variable . However, when we retrieve it from `entry.second`, the object is of type `PlanNodeStats`. Unfortunately, this object does not contain the `kNumWrittenFiles `and `kPhysicalWrittenBytes` information for TableWrite.

Reply:
Why do we put this loop within `for (const auto& entry : status.operatorStats) {`? Seems they are not relevant.

Reply:
@ulysses-you @rui-mo 
The planStats is as follows:

```
-- TableWrite[] -> rows:BIGINT, fragments:VARBINARY, commitcontext:VARBINARY
   Output: 2 rows (48.19KB, 1 batches), Cpu time: 13.89ms, Blocked wall time: 0ns, Peak memory: 1.00MB, Memory allocations: 0
      numWrittenFiles              sum: 1, count: 1, min: 1, max: 1
      queuedWallNanos              sum: 4.00us, count: 1, min: 4.00us, max: 4.00us
      runningAddInputWallNanos     sum: 4.22ms, count: 1, min: 4.22ms, max: 4.22ms
      runningFinishWallNanos       sum: 0ns, count: 1, min: 0ns, max: 0ns
      runningGetOutputWallNanos    sum: 9.66ms, count: 1, min: 9.66ms, max: 9.66ms
  -- Project[expressions: (n1_2:BIGINT, cast "n0_0" as BIGINT), (n1_3:BIGINT, cast "n0_1" as BIGINT)] -> n1_2:BIGINT, n1_3:BIGINT
     Output: 1 rows (64B, 1 batches), Cpu time: 940.70us, Blocked wall time: 0ns, Peak memory: 256B, Memory allocations: 2
        runningAddInputWallNanos     sum: 5.46us, count: 1, min: 5.46us, max: 5.46us
        runningFinishWallNanos       sum: 2.81us, count: 1, min: 2.81us, max: 2.81us
        runningGetOutputWallNanos    sum: 928.96us, count: 1, min: 928.96us, max: 928.96us
    -- ValueStream[] -> n0_0:INTEGER, n0_1:INTEGER
       Input: 0 rows (0B, 0 batches), Output: 1 rows (64B, 1 batches), Cpu time: 1.04ms, Blocked wall time: 0ns, Peak memory: 0B, Memory allocations: 0
          runningAddInputWallNanos     sum: 0ns, count: 1, min: 0ns, max: 0ns
          runningFinishWallNanos       sum: 1.60us, count: 1, min: 1.60us, max: 1.60us
          runningGetOutputWallNanos    sum: 1.04ms, count: 1, min: 1.04ms, max: 1.04ms

```
However, we can only retrieve the `numWrittenFiles ` information from the PlanNodeStats. The numWrittenBytes is not recorded, so we are unable to obtain it. Consequently, I have removed the `numWrittenBytes `and will only display the `numWrittenFiles`.

Reply:
According to https://github.com/facebookincubator/velox/blob/main/velox/exec/TableWriter.cpp#L247, `numWrittenBytes` will be reflected in `physicalWrittenBytes`, could you confirm? Thanks.

Reply:
Yes. However the `physicalWrittenBytes ` is not recorded in `PlanNodeStats`.

Reply:
Seems it can be accessed through operator stats just like `outputVectors`, no?
https://github.com/facebookincubator/velox/blob/main/velox/exec/Operator.h#L123

Reply:
@rui-mo @ulysses-you Filed https://github.com/facebookincubator/velox/pull/8213 to add the physicalWrittenBytes into the PlanNodeStats. Please help to review again. Thanks.